packages/jest-core/src/SearchSource.ts

```diff
@@ -111,9 +111,37 @@ export default class SearchSource {
 
     const testCases = Array.from(this._testPathCases); // clone
     if (testPathPattern) {
-      const regex = testPathPatternToRegExp(testPathPattern);
+      let amendedTestPathPattern = testPathPattern;
+
+      const pathSeperator = process.platform === 'win32' ? '\\\\' : path.sep;
```

Comment:
This shouldn't be necessary. Isn't the point of `path.sep` for cross-platform support of different path separators?

Reply:
path.sep returns \\ but that’s needs to get escaped when it’s passed to the new RegExp call. There’s actually a similar function in normalize.ts under jest-config that does this but isn’t exported.